lib/carto/oauth_provider/scopes/default_scope.rb

```diff
@@ -10,7 +10,11 @@ def initialize(type, service, category, description)
 
         def grant_section(grants)
           section = grants.find { |i| i[:type] == @type }
-          section || { type: @type, @grant_key => [] }
+          section = section || { type: @type, @grant_key => [] }
+          if not section.key?(@grant_key)
+            section[@grant_key] = []
+          end
```

Comment:
A more idiomatic way, although it's not exactly the same. This is adding the empty array to the hash when the key is not present or the value is nil. I guess it's valid anyway.

```suggestion
          section[@grant_key] ||= []
```